libraries/AP_Scripting/applets/Script_Controller.lua

```diff
@@ -0,0 +1,185 @@
+--[[
+   a script to select other lua scripts using an auxillary switch
+--]]
+
+local THIS_SCRIPT = "Script_Controller.lua"
+
+local scripting_dir_select = assert(rc:find_channel_for_option(302), 'no sw assigned to 302 for script directory selection') -- trick script selection channel
+
+--[[
+   copy file src to dest, return true on success
+--]]
+function file_copy(src, dest)
+   local block_size = 256
+   local file1 = io.open(src, "rb")
+   if not file1 then
+      return false
+   end
+   local file2 = io.open(dest, "wb")
+   if not file2 then
+      file1:close()
+      return false
+   end
+   while true do
+      local block = file1:read(block_size)
+      if not block then
+         break
+      end
+      file2:write(block)
+   end
+   local ret = false
+   if file1:seek("end") == file2:seek("end") then
+      ret = true
+   end
+   file1:close()
+   file2:close()
+   return ret
+end
+
+--[[
+   compare two files, return true if they are the same
+--]]
+function file_compare(filename1, filename2)
+   local block_size = 256
+   local file1 = io.open(filename1, "rb")
+   if not file1 then
+      return false
+   end
+   local file2 = io.open(filename2, "rb")
+   if not file2 then
+      file1:close()
+      return false
+   end
+   local ret = true
+   while true do
+      local block1 = file1:read(block_size)
+      local block2 = file2:read(block_size)
+      if block1 ~= block2 then
+         ret = false
+         break
+      end
+      if not block1 then
+         break
+      end
+   end
+   file1:close()
+   file2:close()
+   return ret
+end
+
+--[[
+   get the path to the scripts directory. This will be scripts/ on SITL
+   and APM/scripts on a ChibiOS board
+--]]
+function get_scripts_dir()
+   local dlist1 = dirlist("APM/scripts")
+   if dlist1 and #dlist1 > 0 then
+      return "APM/scripts"
+   end
+   -- otherwise assume scripts/
+   return "scripts"
+end
+
+--[[
+    check if a file exists, returns true if it does' fname is complete path to file
+--]]
+function file_exists(fname)
+   local f = io.open(fname,"rb")
+   if not f then
+      return false
+   end
+   f:close()
+   return true
+end
+
+--[[
+   remove any lua scripts in the scripts directory that are not in the given subdir besides ourselves
+   returns true if any files were removed
+--]]
+function remove_scripts(subdir)
+   local sdir = get_scripts_dir()
+   local dlist = dirlist(sdir)
+   if not dlist then
+      return false
+   end
+   local ret = false
+   for k,v in ipairs(dlist) do
+      local suffix = v:sub(-4)
+      if suffix == ".lua" and v ~= THIS_SCRIPT then
```

Comment:
we need to make these comparisons case-insensitive with the upper() method

Reply:
done